Source/Extensions/Blazorise.BusyLoading/BusyLoading.razor.cs

```diff
@@ -0,0 +1,216 @@
+            {
+                isLoaded = val;
+                IsLoadedChanged.InvokeAsync( val );
+                InvokeAsync( StateHasChanged );
+            }
+        }
+
+        private void Service_BusyChanged( bool val ) => Busy( val );
+        private void Service_LoadedChanged( bool val ) => Loaded( val );
+
+        private bool GetIsBusy() => isBusy ?? IsBusy;
+        private bool GetIsLoaded() => isLoaded ?? IsLoaded;
+
+        /// <inheritdoc/>
+        public override Task SetParametersAsync( ParameterView parameters )
+        {
+            if ( parameters.TryGetValue( nameof( IsLoaded ), out bool newIsLoadedParameter ) )
+            {
+                if ( isLoadedParameter != newIsLoadedParameter )
+                {
+                    isLoaded = null; // use parameter instead of local value
+                }
+            }
+
+            if ( parameters.TryGetValue( nameof( IsBusy ), out bool newIsBusyParameter ) )
+            {
+                if ( isBusyParameter != newIsBusyParameter )
+                {
+                    isBusy = null; // use parameter instead of local value
+                }
+            }
+
+            return base.SetParametersAsync( parameters );
+        }
+
+        void IDisposable.Dispose()
+        {
+            Service = null;
+        }
+
+        #endregion
+
+        #region Properties
+
+        /// <summary>
+        /// Service used to control this instance
+        /// </summary>
+        [Parameter]
+        public BusyLoadingService Service
+        {
+            get => service;
+            set
+            {
+                if ( value != service )
+                {
+                    if ( service != null )
+                    {
+                        service.BusyChanged -= Service_BusyChanged;
+                        service.LoadedChanged -= Service_LoadedChanged;
+                    }
+
+                    service = value;
+                    if ( service != null )
+                    {
+                        service.BusyChanged += Service_BusyChanged;
+                        service.LoadedChanged += Service_LoadedChanged;
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Indicates whether component is ready to be rendered
+        /// </summary>
+        [Parameter]
+        public bool IsLoaded
+        {
+            get => isLoaded ?? isLoadedParameter;
+            set => isLoadedParameter = value;
+        }
+
+        /// <summary>
+        /// Indicates whether the component should be covered with a busy screen
+        /// </summary>
+        [Parameter]
+        public bool IsBusy
+        {
+            get => isBusy ?? isBusyParameter;
+            set => isBusyParameter = value;
+        }
+
+        /// <summary>
+        /// Occurs when IsLoaded state has changed
+        /// </summary>
+        [Parameter]
+        public EventCallback<bool> IsLoadedChanged { get; set; }
```

Comment:
As per put [Contributing guidelines](https://github.com/Megabit/Blazorise/wiki/Contributing), you can remove the `Is` prefix.